chirp/drivers/icf520.py

```diff
@@ -0,0 +1,1604 @@
+        _group.append(RadioSetting(
+            "timer_exo_delay", "EXO Delay Timer",
+            RadioSettingValueFloat(0, 25.5, _timer_exo, precision=1)))
+        _group.append(RadioSetting(
+            "timer_horn", "EXO/Horn Timer",
+            RadioSettingValueInteger(0, 255, self._memobj.timer.horn)))
+        _group.append(RadioSetting(
+            "scrambler_non_rolling", "Scrambler Type",
+            RadioSettingValueList(
+                SCRAMBLER_TYPE_VALUES, current_index=int(
+                    self._memobj.scrambler_non_rolling))))
+        _group.append(RadioSetting(
+            "scrambler_group_code", "Scrambler Group Code",
+            RadioSettingValueList(
+                SCRAMBLER_GCODE_VALUES, current_index=int(
+                    self._memobj.scrambler_group_code))))
+        _group.append(RadioSetting(
+            "scrambler_syn_capture", "Scrambler Synchronous Capture",
+            RadioSettingValueList(
+                SYN_CAPTURE_CHOICE, current_index=sc_sf(
+                    self._memobj.scrambler_syn_capture))))
+        _group.append(RadioSetting(
+            "scrambler_tone_start_timing", "Scrambler Tone Start Timing",
+            RadioSettingValueList(
+                TONE_STIMING_VALUES, current_index=int(
+                    self._memobj.scrambler_tone_start_timing))))
+        _group.append(RadioSetting(
+            "timer_emergency_switch_on", "Emergency Switch On Timer",
+            RadioSettingValueFloat(0, 25.5, _timer_es_on, precision=1)))
+        _group.append(RadioSetting(
+            "timer_emergency_switch_off", "Emergency Switch Off Timer",
+            RadioSettingValueFloat(0, 25.5, _timer_es_off, precision=1)))
+        _group.append(RadioSetting(
+            "emergency_start_repeat", "Emergency Start/Repeat",
+            RadioSettingValueInteger(
+                0, 255, self._memobj.timer.emergency_start_repeat)))
+        return _group
+
+    # Build the top menu.
+    def get_settings(self):
+        """Returns a RadioSettingGroup containing one or more
+        RadioSettingGroup or RadioSetting objects. These represent general
+        setting knobs and dials that can be adjusted on the radio."""
+        return list(
+            RadioSettingGroup(
+                "top", "LMR",
+                self._get_continuous_tone_group(),
+                self._get_scan_group(),
+                self._get_conv_key_group(),
+                self._get_trunk_key_group(),
+                self._get_display_group(),
+                self._get_set_mode_group(),
+                self._get_common_group()))
+
+    def set_settings(self, settings):
+        """Accepts the top-level RadioSettingGroup returned from get_settings()
+        and adjusts the values in the radio accordingly. This function expects
+        the entire RadioSettingGroup hierarchy returned from get_settings()."""
+        for element in settings:
```

Comment:
Also check out `MemSetting` which takes a symbolic path into the memory object and can set boolean, integer, string, and map types directly so you don't have to manually do everything in `set_settings()` yourself.